File: Binary Tree/104. Maximum Depth of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {

    return findNode(root);

    function findNode(node) {
        if (node == null) {
            return 0;
        }

        var nodeLeftDepth = 1;
        var nodeRightDepth = 1;

        if (node.left != null) {
            nodeLeftDepth = nodeLeftDepth + findNode(node.left);
        }

        if (node.right != null) {
            nodeRightDepth = nodeRightDepth + findNode(node.right);
        }

        if (nodeLeftDepth > nodeRightDepth) {
            return nodeLeftDepth;
        } else {
            return nodeRightDepth;
        }
    }
};

File: NeetCode 150/104. Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        self.max_level = 0

        def dfs(node, level):
            if not node:
                return level
            
            return max(dfs(node.left, level + 1), dfs(node.right, level + 1))

        return dfs(root, 0)